enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,358 @@
+* Extract all configuration from after the source config point to final destination into the collector configuration.
+
+### Test Plan
+#### Unit testing
+* Log forwarding will add unit tests to provide adequate coverage for all changes
+* BDD unit testing will be added to unit testing to make tests goals more explicit,  readable, and obvious
+#### Integration and E2E tests
+* Test no regressions after upgrade when log forwarding exists
+* Tests to verify there are no regressions in functionality between pre and post logforwarding feature
+* Tests to verify log forwarding is writing logs to an Elasticsearch instance not managed by cluster logging
+* Tests to verify log forwarding is writing logs to an Elasticsearch instance managed by cluster logging
+* Tests to verify log forwarding is writing logs to a Fluentd instance via secure forward that is not managed by cluster logging
+
+### Graduation Criteria
+
+##### Tech Preview
+
+* Ability to store logs to the cluster managed log aggregator without defining log forwarding (e.g. status quo)
+* Explicitly define and document log sources and their meanings (e.g. `logs.apps`, `logs.infra`)
+* Explicitly define supported outputs (e.g. `elasticsearch`, `fluentd`)
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Concrete API
+- Additional supported outputs (e.g. `Splunk`, `Graylog`, `kafka`)
+- Support for SRE usecase: As an administrator of a dedicated cluster, I need to allow dedicated-admins to configure the destination of container logs (logs.apps) separately from infra logs so they are prevented from modifying where infra logs (logs.infra) are forwarded
+- Modify API to support: include and exclude of logs
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+Upgrades should be seamless to users of cluster logging deployments managed by the `cluster-logging-operator`.  The current implementation is a static file mounted into the collector daemonset.  Currently users must
+explicitly identify a deployment as 'Unmanaged' in order to make changes to the collector's configuration. Admins who make no changes will see no difference after upgrade when this feature is deployed.
+Upon upgrade, the `cluster-logging-operator` will create a `spec.forwarding` section routing both `logs.infra` and `logs.app` logs to the internal logstore if the logstore is defined in the `clusterlogging` resource and
+`spec.forwarding` is not.  The `cluster-logging-operator` will set a status field to indicate it performed the upgrade and will take no further action duing subsequent migrations or `cluster-loging-operator` restarts.
```

Comment:
fyi This migration path was chosen due to OLM not supporting conversion webhooks today.